### PR TITLE
home-manager: enable nix run for activationPackage

### PR DESCRIPTION
**Pull request by ant-arctica**
### Description

Adds a symlink from `$out/activate` to `$out/bin/home-manager-generation` in the home-manager-generation activationPackage.
This makes it possible for people who put their home-manager configuration in a flake (not as a part of a NixOS configuration)
to update their home with: 
```nix run flake#homeMangerConfigurations.<name>.activationPackage```
Instead of  with `nix build <stuff> && ./result/activate` which is more cumbersome, and pollutes your `gcroots/auto` with stray symlinks to your current folder.

### Checklist

- [x] Change is backwards compatible.

- [x] Code formatted with `./format`.

- [x] Code tested through `nix-shell --pure tests -A run.all`.
    
[  ] Test cases updated/added. See [example](https://github.com/nix-community/home-manager/commit/f3fbb50b68df20da47f9b0def5607857fcc0d021#diff-b61a6d542f9036550ba9c401c80f00ef).
  * I don't think there are tests for activationPackage

- [x] Commit messages are formatted like

    ```
    {component}: {description}

    {long description}
    ```

    See [CONTRIBUTING](https://github.com/nix-community/home-manager/blob/master/doc/contributing.adoc#sec-commit-style) for more information and [recent commit messages](https://github.com/nix-community/home-manager/commits/master) for examples.

- If this PR adds a new module

  - [ ] Added myself as module maintainer. See [example](https://github.com/nix-community/home-manager/blob/068ff76a10e95820f886ac46957edcff4e44621d/modules/programs/lesspipe.nix#L6).

  - [ ] Added myself and the module files to `.github/CODEOWNERS`.

### CODE REVIEW

File: modules/home-environment.nix [hunk 1]
```diff
@@ -552,6 +552,9 @@ in
 
             cp ${activationScript} $out/activate
 
+            mkdir $out/bin
+            ln -s $out/activate $out/bin/home-manager-generation
```
Comment (berbiche) [inline modules/home-environment.nix:556]:
```suggestion
            ln -s $out/activate $out/bin/activate
```

To have a similar API to NixOS.

**Review by berbiche — COMMENTED** [review]
Thanks for the contribution :+1: 

This will be quite useful to many users.

File: modules/home-environment.nix (hunk 1 above)
Comment (ant-arctica) [reply modules/home-environment.nix:556]: When I tested it, nix run only tries to run `$out/bin/<package-name>` (that's why `nix run nixpkgs#neovim` doesn't work).
So if the symlink is moved to a different location you can't run it automatically anymore.

Comment (ant-arctica) [reply modules/home-environment.nix:556]: see NixOS/nix#4498

Comment (berbiche) [reply modules/home-environment.nix:556]: Right, I forgot about this peculiarity of `nix run`. It's good to know `meta.mainProgram` exists now.
I usually always do `nix shell nixpkgs#something -c some-program`

**Comment by ant-arctica** [comment]
Closed in favor of #1856

**Pull request closed without merging**